samples/dubbogo/simple/nacos/server/app/user.go

```diff
@@ -0,0 +1,271 @@
+func (u *UserProvider) CreateUser(ctx context.Context, user *User) (*User, error) {
+	outLn("Req CreateUser data:%#v", user)
+	if user == nil {
+		return nil, errors.New("not found")
+	}
+	_, ok := cache.GetByName(user.Name)
+	if ok {
+		return nil, errors.New("data is exist")
+	}
+
+	b := cache.Add(user)
+	if b {
+		return user, nil
+	}
+
+	return nil, errors.New("add error")
+}
+
+// GetUserByName query by name, single param, PX config GET.
+func (u *UserProvider) GetUserByName(ctx context.Context, name string) (*User, error) {
+	outLn("Req GetUserByName name:%#v", name)
+	r, ok := cache.GetByName(name)
+	if ok {
+		outLn("Req GetUserByName result:%#v", r)
+		return r, nil
+	}
+	return nil, nil
+}
+
+// GetUserByCode query by code, single param, PX config GET.
+func (u *UserProvider) GetUserByCode(ctx context.Context, code int64) (*User, error) {
+	outLn("Req GetUserByCode name:%#v", code)
+	r, ok := cache.GetByCode(code)
+	if ok {
+		outLn("Req GetUserByCode result:%#v", r)
+		return r, nil
+	}
+	return nil, nil
+}
+
+// GetUserTimeout query by name, will timeout for pixiu.
+func (u *UserProvider) GetUserTimeout(ctx context.Context, name string) (*User, error) {
+	outLn("Req GetUserByName name:%#v", name)
+	// sleep 10s, pixiu config less than 10s.
+	time.Sleep(10 * time.Second)
+	r, ok := cache.GetByName(name)
+	if ok {
+		outLn("Req GetUserByName result:%#v", r)
+		return r, nil
+	}
+	return nil, nil
+}
+
+// GetUserByNameAndAge query by name and age, two params, PX config GET.
+func (u *UserProvider) GetUserByNameAndAge(ctx context.Context, name string, age int32) (*User, error) {
+	outLn("Req GetUserByNameAndAge name:%s, age:%d", name, age)
+	r, ok := cache.GetByName(name)
+	if ok && r.Age == age {
+		outLn("Req GetUserByNameAndAge result:%#v", r)
+		return r, nil
+	}
+	return r, nil
+}
+
+// UpdateUser update by user struct, my be another struct, PX config POST or PUT.
+func (u *UserProvider) UpdateUser(ctx context.Context, user *User) (bool, error) {
+	outLn("Req UpdateUser data:%#v", user)
+	r, ok := cache.GetByName(user.Name)
+	if ok {
+		if user.ID != "" {
+			r.ID = user.ID
+		}
+		if user.Age >= 0 {
+			r.Age = user.Age
+		}
+		return true, nil
+	}
+	return false, errors.New("not found")
+}
+
+// UpdateUserByName update by user struct, my be another struct, PX config POST or PUT.
+func (u *UserProvider) UpdateUserByName(ctx context.Context, name string, user *User) (bool, error) {
+	outLn("Req UpdateUserByName data:%#v", user)
+	r, ok := cache.GetByName(name)
+	if ok {
+		if user.ID != "" {
+			r.ID = user.ID
+		}
+		if user.Age >= 0 {
+			r.Age = user.Age
+		}
+		return true, nil
+	}
+	return false, errors.New("not found")
+}
+
+// nolint
+func (u *UserProvider) Reference() string {
+	return "UserProvider"
+}
+
+// nolint
+func (u User) JavaClassName() string {
+	return "com.dubbogo.pixiu.User"
+}
+
+// nolint
+func outLn(format string, args ...interface{}) {
+	fmt.Printf("\033[32;40m"+format+"\033[0m\n", args...)
```

Comment:
If u really wannt use the color prefix, pls define them as const in common/color.go. Follow the following example from my c lib lisk

```go
   static const n1* color_string_[] = {
        [ESCREEN_COLOR_MIN] = "",
        [ENORMAL] = "\033[0m",
        [EBLUE] = "\033[1;34;40m",
        [ERED] = "\033[1;31;40m",
        [EPRUPLE] = "\033[1;35;40m",
        [EWHITE] = "\033[1;37;40m",
        [EGREEN] = "\033[1;32;40m",
        [EAQUA] = "\033[1;36;40m",
        [EHIGH_LIGHT] = "\033[1;48;40m",
        [EYELLOW] = "\033[1;33;40m",
        [EUNDER_LINE] = "\033[1;4;40m",
        [EUNDER_LINE_TWINKLE] = "\033[1;5;40m",
        [EUNDER_LINE_TWINKLE_HIGH_LIGHT] = "\033[1;7;40m"
    };
```

Reply:
delete outLn function in all samples